File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program19.java
//////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept number from user and return multiplication of all digits.
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class Solution
{
	public int iValue;

	public Solution(int iValue)
	{
		this.iValue = iValue;
	}

	public int MultiDigits()
	{
                      	
		int iCnt=0,iMulti=1;
		while(iValue!=0)
		{
			int iDigit = iValue%10;
			iMulti=iMulti*iDigit;
			iValue = iValue/10;
		}        
		return iMulti;
	
	}
}

class Program
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		int iRet=0;
		System.out.println("Enter number: ");
		int iNo = sObj.nextInt();

		Solution obj = new Solution(iNo);
		iRet = obj.MultiDigits();
	
		System.out.println("Multiplication is: "+iRet);
	}
}

/*
OUTPUT:

Enter number:
2395
Multiplication is: 270

*/

File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program19.cpp
//////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept number from user and return multiplication of all digits.
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Display
{
	private:
		int iNo;

	public:

		Display(int iNo)
		{
			this->iNo = iNo;
		}	
		
		int MultiDigits()
		{
			int iDigit = 0,iMulti=1;

			if(iNo<0)
			{
				iNo = -iNo;
			}

			while(iNo>0)
			{
				iDigit = iNo%10;
				if(iDigit>0)
				{
					iMulti = iMulti*iDigit;
				}	
				iNo = iNo/10;
			}

			return iMulti;
		}
};

int main()
{
	int iValue=0,iRet=0;

	cout<<"Enter Number: ";
	cin>>iValue;

	Display *dObj = new Display(iValue);
	iRet = dObj->MultiDigits();

	cout<<"Multiplication of Digits are:  "<<iRet<<"\n";
	return 0;
}

/*
OUTPUT:

Enter Number: 2395
Multiplication of Digits are:  270

*/
